src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -717,6 +769,18 @@ namespace vcpkg
         from_env(get_env, ASSET_SOURCES_ENV, asset_sources_template_env);
         from_env(get_env, REGISTRIES_CACHE_DIR_ENV, registries_cache_dir);
 
+        // detect whether we are running in a CI environment
+        for (auto&& ci_env_var : get_known_ci_variables())
+        {
+            Optional<std::string> maybe_val;
+            from_env(get_env, ci_env_var.first, maybe_val);
+            if (maybe_val.has_value())
```

Comment:
```suggestion
            if (get_env(ci_env_var.first).has_value())
```

`from_env()` implements the "fallback logic" of command line >> environment -- if we don't have a CLI flag, we don't need to use it.